fvm/state/state.go

```diff
@@ -15,37 +12,34 @@ const (
 	DefaultMaxInteractionSize = 2_000_000_000 // ~2GB
 )
 
-type StateOption func(st *State) *State
-
 type State struct {
-	ledger Ledger
-	// reason for using mutex instead of channel
-	// state doesn't have to be thread safe and right now
-	// is only used in a single thread but a mutex has been added
-	// here to prevent accidental multi-thread use in the future
-	lock                  sync.Mutex
-	draft                 map[string]payload
+	ledger                Ledger
+	parent                *State
```

Comment:
I'm not sure if we need a hierarchy of states. State works as a wrapper for view with extra operation which only apply on one level (like fees)
We should probably just keep `delta.View` hierarchy here

Reply:
fixed.